src/Files.App/Data/Items/ShelfItem.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Files.Shared.Utils;
+
+namespace Files.App.Data.Items
+{
+	[Bindable(true)]
+	public sealed partial class ShelfItem : ObservableObject, IWrapper<IStorable>, IAsyncInitialize
+	{
+		private readonly IImageService _imageService;
+
+		[ObservableProperty] private IImage? _Icon;
+		[ObservableProperty] private string? _Name;
+		[ObservableProperty] private string? _Path;
+
+		/// <inheritdoc/>
+		public IStorable Inner { get; }
+
+		public ShelfItem(IStorable storable, IImage? icon = null)
+		{
+			_imageService = Ioc.Default.GetRequiredService<IImageService>();
+			Inner = storable;
+			Icon = icon;
+			Name = storable.Name;
+			Path = storable.Id;
+		}
+
+		/// <inheritdoc/>
+		public async Task InitAsync(CancellationToken cancellationToken = default)
+		{
+			Icon = await _imageService.GetIconAsync(Inner, cancellationToken);
```

Comment:
Are we still using image service? I believe we've switched to `FileThumbnailHelper.GetIconAsync`.

Reply:
`FileThumbnailHelper.GetIconAsync` returns a `byte[]` array that is hard to work with and would break (or require workarounds) in the view model to support images. `ImageService`, on the other hand, returns an `IImage` that already contains a BitmapImage ready to be parsed and extracted by the UI side. It uses FileThumbnailHelper in the underlying implementation anyway